### ISSUE TITLE
Shadowmask overlap detection slightly broken

### ISSUE THREAD

**Issue by jonathanrpace** [guycalledfrank/bakery-issues#145]
TLDR 
`ConvexIntersect(SpotlightA, SpotlightB) != ConvexIntersect(SpotlightB, SpotlightA)`

I noticed that Bakery would halt and inform the user about some shadowmasked lights not being able to be baked because more than 4 overlap.

Looking at the light cones in my scene, it seemed that it was probably being a bit too cautious. So I wrote some debug visualisation that uses your ConvexIntersect code to draw gizmos for lights overlapping the selected one.

It became clear that ConvexIntersect() was returning different results depending on which order I supplied the parameters.

I have a local fix - effectively replacing:
`if ( ConvexIntersect(A, B) ) overlap = true`
with
`if ( ConvexIntersect(A, B) && ConvexIntersect(B, A) ) overlap = true`
This is inside ftRenderLightmap

**Comment by guycalledfrank** [comment]
That's a super good catch, thanks!

Patched: https://github.com/guycalledfrank/bakery-csharp/commit/7ab81c5f5cc76e36d0951dfc380de1d105bb518e

At first I thought there might be some bug with my convex normals/vertices, so I made a little test project with debug visualization, but they looked alright... really it's just a necessary part of the algorithm to check it both ways, otherwise it's inaccurate, but I didn't notice, because I didn't have a debug vis.
